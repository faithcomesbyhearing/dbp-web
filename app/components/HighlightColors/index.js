/**
 *
 * HighlightColors
 *
 */

import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';
import Colors from '../../../theme_config/javascriptColors';
import messages from './messages';

function HighlightColors({ addHighlight }) {
	return (
		<>
			<span className={'color-group'}>
				{}
				<span
					role={'button'}
					tabIndex={0}
					className={'none'}
					onClick={(e) =>
						addHighlight({
							popupCoords: { x: e.clientX, y: e.clientY },
							color: 'none',
						})
					}
				/>
				<span className={'color-text'}>
					<FormattedMessage {...messages.none} />
				</span>
			</span>
			<span className={'color-group'}>
				{}
				<span
					role={'button'}
					tabIndex={-1}
					className={'yellow'}
					onClick={(e) =>
						addHighlight({
							popupCoords: { x: e.clientX, y: e.clientY },
							color: Colors.highlightYellow,
						})
					}
				/>
				<span className={'color-text'}>
					<FormattedMessage {...messages.yellow} />
				</span>
			</span>
			<span className={'color-group'}>
				{}
				<span
					role={'button'}
					tabIndex={-2}
					className={'green'}
					onClick={(e) =>
						addHighlight({
							popupCoords: { x: e.clientX, y: e.clientY },
							color: Colors.highlightGreen,
						})
					}
				/>
				<span className={'color-text'}>
					<FormattedMessage {...messages.green} />
				</span>
			</span>
			<span className={'color-group'}>
				{}
				<span
					role={'button'}
					tabIndex={-3}
					className={'pink'}
					onClick={(e) =>
						addHighlight({
							popupCoords: { x: e.clientX, y: e.clientY },
							color: Colors.highlightPink,
						})
					}
				/>
				<span className={'color-text'}>
					<FormattedMessage {...messages.pink} />
				</span>
			</span>
			<span className={'color-group'}>
				{}
				<span
					role={'button'}
					tabIndex={-4}
					className={'purple'}
					onClick={(e) =>
						addHighlight({
							popupCoords: { x: e.clientX, y: e.clientY },
							color: Colors.highlightPurple,
						})
					}
				/>
				<span className={'color-text'}>
					<FormattedMessage {...messages.purple} />
				</span>
			</span>
			<span className={'color-group'}>
				{}
				<span
					role={'button'}
					tabIndex={-5}
					className={'blue'}
					onClick={(e) =>
						addHighlight({
							popupCoords: { x: e.clientX, y: e.clientY },
							color: Colors.highlightBlue,
						})
					}
				/>
				<span className={'color-text'}>
					<FormattedMessage {...messages.blue} />
				</span>
			</span>
		</>
	);
}

HighlightColors.propTypes = {
	addHighlight: PropTypes.func,
};

export default HighlightColors;
