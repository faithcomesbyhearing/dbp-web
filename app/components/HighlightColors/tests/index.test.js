import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from '@cfaester/enzyme-adapter-react-18';
import Colors from '../../../../theme_config/javascriptColors';
import HighlightColors from '..';

Enzyme.configure({ adapter: new Adapter() });

/* eslint-disable react/prop-types */
jest.mock('react-intl', () => ({
	FormattedMessage: ({ defaultMessage }) => <span>{defaultMessage}</span>,
	defineMessages: (messages) => messages,
}));
/* eslint-enable react/prop-types */
const options = ['None', 'Yellow', 'Green', 'Pink', 'Purple', 'Blue'];
const green = Colors.highlightGreen;
const yellow = Colors.highlightYellow;
const pink = Colors.highlightPink;
const purple = Colors.highlightPurple;
const blue = Colors.highlightBlue;
let addHighlight = jest.fn(() => {});
let wrapper;

describe('HighlightColors component', () => {
	beforeEach(() => {
		addHighlight = jest.fn(({ color }) => color);
		wrapper = Enzyme.mount(<HighlightColors addHighlight={addHighlight} />);
	});
	it('should match snapshot', () => {
		const tree = renderer.create(
			<HighlightColors addHighlight={addHighlight} />,
		);
		expect(tree).toMatchSnapshot();
	});
	it('should contain yellow highlight option', () => {
		const colorNode = wrapper.find('.yellow');
		const spy = jest.spyOn(colorNode.props(), 'onClick');
		const colorOutput = colorNode.props().onClick({ clientX: 10, clientY: 10 });

		expect(spy).toHaveBeenCalledTimes(1);
		expect(colorOutput).toEqual(yellow);

		colorNode.simulate('click');

		expect(addHighlight).toHaveBeenCalledTimes(2);
	});
	it('should contain green highlight option', () => {
		const colorNode = wrapper.find('.green');
		const spy = jest.spyOn(colorNode.props(), 'onClick');
		const colorOutput = colorNode.props().onClick({ clientX: 10, clientY: 10 });

		expect(spy).toHaveBeenCalledTimes(1);
		expect(colorOutput).toEqual(green);

		colorNode.simulate('click');

		expect(addHighlight).toHaveBeenCalledTimes(2);
	});
	it('should contain pink highlight option', () => {
		const colorNode = wrapper.find('.pink');
		const spy = jest.spyOn(colorNode.props(), 'onClick');
		const colorOutput = colorNode.props().onClick({ clientX: 10, clientY: 10 });

		expect(spy).toHaveBeenCalledTimes(1);
		expect(colorOutput).toEqual(pink);

		colorNode.simulate('click');

		expect(addHighlight).toHaveBeenCalledTimes(2);
	});
	it('should contain purple highlight option', () => {
		const colorNode = wrapper.find('.purple');
		const spy = jest.spyOn(colorNode.props(), 'onClick');
		const colorOutput = colorNode.props().onClick({ clientX: 10, clientY: 10 });

		expect(spy).toHaveBeenCalledTimes(1);
		expect(colorOutput).toEqual(purple);

		colorNode.simulate('click');

		expect(addHighlight).toHaveBeenCalledTimes(2);
	});
	it('should contain blue highlight option', () => {
		const colorNode = wrapper.find('.blue');
		const spy = jest.spyOn(colorNode.props(), 'onClick');
		const colorOutput = colorNode.props().onClick({ clientX: 10, clientY: 10 });

		expect(spy).toHaveBeenCalledTimes(1);
		expect(colorOutput).toEqual(blue);

		colorNode.simulate('click');

		expect(addHighlight).toHaveBeenCalledTimes(2);
	});
	it('should contain blue highlight option', () => {
		const colorNode = wrapper.find('.none');
		const spy = jest.spyOn(colorNode.props(), 'onClick');
		const colorOutput = colorNode.props().onClick({ clientX: 10, clientY: 10 });

		expect(spy).toHaveBeenCalledTimes(1);
		expect(colorOutput).toEqual('none');

		colorNode.simulate('click');

		expect(addHighlight).toHaveBeenCalledTimes(2);
	});
	it('should contain all the color options', () => {
		const nodes = wrapper.find('.color-text');

		nodes.forEach((node, index) => expect(node.text()).toEqual(options[index]));
	});
});
