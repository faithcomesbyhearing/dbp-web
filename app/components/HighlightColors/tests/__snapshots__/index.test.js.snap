// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HighlightColors component should match snapshot 1`] = `
[
  <span
    className="color-group"
  >
    <span
      className="none"
      onClick={[Function]}
      role="button"
      tabIndex={0}
    />
    <span
      className="color-text"
    >
      <span>
        None
      </span>
    </span>
  </span>,
  <span
    className="color-group"
  >
    <span
      className="yellow"
      onClick={[Function]}
      role="button"
      tabIndex={-1}
    />
    <span
      className="color-text"
    >
      <span>
        Yellow
      </span>
    </span>
  </span>,
  <span
    className="color-group"
  >
    <span
      className="green"
      onClick={[Function]}
      role="button"
      tabIndex={-2}
    />
    <span
      className="color-text"
    >
      <span>
        Green
      </span>
    </span>
  </span>,
  <span
    className="color-group"
  >
    <span
      className="pink"
      onClick={[Function]}
      role="button"
      tabIndex={-3}
    />
    <span
      className="color-text"
    >
      <span>
        Pink
      </span>
    </span>
  </span>,
  <span
    className="color-group"
  >
    <span
      className="purple"
      onClick={[Function]}
      role="button"
      tabIndex={-4}
    />
    <span
      className="color-text"
    >
      <span>
        Purple
      </span>
    </span>
  </span>,
  <span
    className="color-group"
  >
    <span
      className="blue"
      onClick={[Function]}
      role="button"
      tabIndex={-5}
    />
    <span
      className="color-text"
    >
      <span>
        Blue
      </span>
    </span>
  </span>,
]
`;
